src/components/KCatalog/KCatalog.vue

```diff
@@ -74,25 +56,27 @@
     </div>
 
     <div
-      v-else-if="!hasError && (!isCatalogLoading && !isLoading && !isRevalidating) && (data && !data.length)"
-      class="k-catalog-empty-state"
-      data-testid="k-card-catalog-empty-state"
+      v-else-if="!error && (!isCatalogLoading && !loading && !isRevalidating) && (data && !data.length)"
```

Comment:
suggestion: this if/else logic is a bit complicated; can you move into a computed property with comments?